v-next/hardhat-utils/src/ci.ts

```diff
@@ -0,0 +1,6 @@
+/**
+ * Returns 'true' if the current process is running in the GitHub CI environment.
+ */
+export function isCI() {
+  return process.env.CI === "true";
```

Comment:
As we discussed, let's use the same ones than PNPM uses, and let's double check that we cover the same cases than v2, but without any dependency. 

See packages/hardhat-core/src/internal/util/ci-detection.ts

Except for the `isLinuxWithoutDisplayServer()` function that's too restrictive. e.g. it flags devcontainer as a ci.